packages/tpetra/core/src/Tpetra_Details_WrappedDualView.hpp

```diff
@@ -131,11 +132,20 @@ class WrappedDualView {
   { }
 
   WrappedDualView(const DeviceViewType deviceView) {
+    TEUCHOS_TEST_FOR_EXCEPTION(
+        deviceView.data() != nullptr && deviceView.use_count() == 0,
+        std::invalid_argument,
+        "Tpetra::Details::WrappedDualView: cannot construct with a device view that\n"
+        "does not own its memory (i.e. constructed with a raw pointer and dimensions)\n"
+        "because the WrappedDualView needs to assume ownership of the memory.");
+    //If the provided view is default-constructed (null, 0 extent, 0 use count),
+    //leave the host mirror default-constructed as well in order to have a matching use count of 0.
     HostViewType hostView;
-    if (deviceView.data() != nullptr) {
+    if(deviceView.use_count() != 0)
```

Comment:
Thanks for fixing this, Brian!  Can you also add unit tests to cover this logic so we don't forget why it's here and take it out later?  Also can you add a unit test ensuring the exception will be thrown under the right conditions?

Reply:
@tasmith4 I just added these. For the first one, if you construct a normal Kokkos::View with a label and zero extent, it never ends up with ``data() == nullptr``. But with ``View<Sacado::MP::Vector>``, it does - there is a lot of custom View code in there that I don't fully understand. So that test had to go in Stokhos.